src/qamqpexchange.cpp

```diff
@@ -188,6 +218,47 @@ void QAmqpExchangePrivate::handleAckOrNack(const QAmqpMethodFrame &frame)
     }
 }
 
+/*! Report and change state. */
+void QAmqpExchangePrivate::newState(ExchangeState state)
+{
+    qAmqpDebug() << "Exchange state: "
```

Comment:
these are going to be great entry points for the as-of-now terrible logs :smile:, I think perhaps it makes sense to make this a little more informative and condensed: ```qAmqpDebug() << "exchange(" << howeverWeGetExchangeName() << exchangeState << " => " << state;```